src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisShardSyncer.java

```diff
@@ -354,6 +363,44 @@ private synchronized void assertHashRangeOfClosedShardIsCovered(Shard closedShar
         return shards;
     }
 
+    private List<Shard> getShardListAtInitialPosition(IKinesisProxy kinesisProxy,
+                                                      InitialPositionInStreamExtended initialPosition)
+            throws KinesisClientLibIOException {
+
+        final ShardFilter shardFilter = getShardFilterAtInitialPosition(initialPosition);
+        final List<Shard> shards = kinesisProxy.getShardListWithFilter(shardFilter);
+
+        if (shards == null) {
```

Comment:
We can also use Optional.orElseThrow() here

Reply:
I chose the null check 1) for consistency with existing code and 2) my personal preference is to avoid optionals on lists, the reasoning being that an "absent" list should be an empty list rather than null. But that's just preference, happy to change it if you'd like :)